LocalFinalVariableName false-positive

cat config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LocalFinalVariableName"/>
  </module>
</module>
```

cat Test.java
```java
import java.util.concurrent.Semaphore;

public class Test {
	private static final Lock LOCK = new Lock();

	public void foo() {
		LOCK.lock();
		try (LOCK) { // violation
		}
	}
}

class Lock implements AutoCloseable {
	private final Semaphore semaphore;

	public Lock() {
		this.semaphore = new Semaphore(1);
	}

	public void lock() {
		try {
			semaphore.acquire();
		} catch (InterruptedException exception) {
			Thread.currentThread().interrupt();
		}
	}

	public void unlock() {
		semaphore.release();
	}

	@Override
	public void close() {
		unlock();
	}
}
```

java -Duser.language=en -Duser.country=US -jar checkstyle-8.36.2-all.jar -c config.xml Test.java
```
Starting audit...
[ERROR] /home/boris/Test.java:8:22: Name 'LOCK' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

`try (LOCK)` should be valid and no warning should be emitted.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java
@@ -266,7 +266,7 @@ public final class ScopeUtil {
         }
 
         if (node.getType() == TokenTypes.RESOURCE) {
-            localVariableDef = true;
+            localVariableDef = node.getChildCount() > 1;
         }
         return localVariableDef;
     }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheckTest.java
@@ -114,4 +114,15 @@ public class LocalFinalVariableNameCheckTest
         verify(checkConfig, getPath("InputLocalFinalVariableNameTryResources.java"), expected);
     }
 
+    @Test
+    public void testTryWithResourcesJava9() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(LocalFinalVariableNameCheck.class);
+        checkConfig.addAttribute("format", "[a-z]+");
+
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getNonCompilablePath(
+            "InputLocalFinalVariableNameTryResourcesJava9.java"), expected);
+    }
+
 }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilTest.java
@@ -166,7 +166,18 @@ public class ScopeUtilTest {
 
     @Test
     public void testIsLocalVariableDefResource() {
-        assertTrue(ScopeUtil.isLocalVariableDef(getNode(TokenTypes.RESOURCE)), "invalid result");
+        final DetailAstImpl node = getNode(TokenTypes.RESOURCE);
+        final DetailAstImpl modifiers = new DetailAstImpl();
+        modifiers.setType(TokenTypes.MODIFIERS);
+        node.addChild(modifiers);
+        final DetailAstImpl ident = new DetailAstImpl();
+        ident.setType(TokenTypes.IDENT);
+        node.addChild(ident);
+        assertTrue(ScopeUtil.isLocalVariableDef(node), "invalid result");
+        final DetailAstImpl resourceWithIdent = getNode(TokenTypes.RESOURCE);
+        resourceWithIdent.addChild(ident);
+        assertFalse(ScopeUtil.isLocalVariableDef(resourceWithIdent), "invalid result");
+        assertFalse(ScopeUtil.isLocalVariableDef(getNode(TokenTypes.RESOURCE)), "invalid result");
     }
 
     @Test

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/localfinalvariablename/InputLocalFinalVariableNameTryResourcesJava9.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/localfinalvariablename/InputLocalFinalVariableNameTryResourcesJava9.java
@@ -1,0 +1,26 @@
+//non-compiled with javac: Compilable with Java9
+package com.puppycrawl.tools.checkstyle.checks.naming.localfinalvariablename;
+
+/*
+ * Config:
+ * format = "^[a-z]+$"
+ */
+public class InputLocalFinalVariableNameTryResourcesJava9 {
+    private static final Lock LOCK = new Lock();
+
+    public void foo() {
+        LOCK.lock();
+        try (LOCK) { // ok
+        }
+    }
+}
+
+class Lock implements AutoCloseable {
+
+    public void lock() {
+    }
+
+    @Override
+    public void close() {
+    }
+}

